### PR TITLE
migrated 2 files from js to ts

### PR DESCRIPTION
**Pull request by shubhangi013**
MIgrated `/src/platform/web/ui/general/StaticView.js` and `/src/platform/web/ui/general/LoadingView.js` from javascript to typescript.
Solves part of #528
proof that the changes run fine, when run locally :

Screenshot from 2021-12-04 23-47-38

### CODE REVIEW

File: src/platform/web/ui/RootView.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ import {LoginView} from "./login/LoginView.js";
 import {SessionLoadView} from "./login/SessionLoadView.js";
 import {SessionPickerView} from "./login/SessionPickerView.js";
 import {TemplateView} from "./general/TemplateView";
-import {StaticView} from "./general/StaticView.js";
+import {StaticView} from "./general/StaticView.ts";
```
Comment (bwindels) [inline src/platform/web/ui/RootView.js]: For typescript, we leave the extension off
```suggestion
import {StaticView} from "./general/StaticView";
```

File: src/platform/web/ui/general/LoadingView.ts [hunk 2]
```diff
@@ -17,7 +17,8 @@ limitations under the License.
 import {TemplateView} from "./TemplateView";
 import {spinner} from "../common.js";
 
-export class LoadingView extends TemplateView {
+export class LoadingView <T> extends TemplateView <T> {
```
Comment (bwindels) [inline src/platform/web/ui/general/LoadingView.ts]: `T` is the type of the view model here, so you can use `any` as the type.
```suggestion
export class LoadingView extends TemplateView<any> {
```

File: src/platform/web/ui/general/LoadingView.ts [hunk 3]
```diff
@@ -17,7 +17,8 @@ limitations under the License.
...
+    className:string;
```
Comment (bwindels) [inline src/platform/web/ui/general/LoadingView.ts]: Not sure what this is for?

File: src/platform/web/ui/general/StaticView.ts [hunk 4]
```diff
@@ -15,15 +15,17 @@ See the License for the specific language governing permissions and
...
-    constructor(value, render = undefined) {
+    private _root?: Element;
+    render: any;
+    constructor(value:any , render : any = undefined) {
         if (typeof value === "function" && !render) {
             render = value;
             value = null;
         }
-        this._root = render ? render(tag, value) : this.render(tag, value);
+        this._root= render ? render(tag, value) : this.render(tag, value);
```
Comment (bwindels) [inline src/platform/web/ui/general/StaticView.ts:28]: Please avoid unrelated cosmetic changes

File: src/platform/web/ui/general/StaticView.ts [hunk 5]
```diff
@@ -15,15 +15,17 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import {tag} from "../general/html";
+import {tag} from "./html";
 
 export class StaticView {
-    constructor(value, render = undefined) {
+    private _root?: Element;
+    render: any;
```
Comment (bwindels) [inline src/platform/web/ui/general/StaticView.ts]:
```suggestion
    render: (typeof tag, T) => Element;
```

File: src/platform/web/ui/general/StaticView.ts [hunk 6]
```diff
@@ -15,15 +15,17 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import {tag} from "../general/html";
+import {tag} from "./html";
 
 export class StaticView {
```
Comment (bwindels) [inline src/platform/web/ui/general/StaticView.ts]:
```suggestion
export class StaticView<T> {
```

With T being the type of the view model used.

**Review by bwindels — COMMENTED** [review]
Thanks for your PR, left some comments

**Comment by shubhangi013** [comment]
> Thanks for your PR, left some comments

Thankyou so much! will do these changes.

File: src/platform/web/ui/general/StaticView.ts (hunk 5 above)
Comment (shubhangi013) [reply src/platform/web/ui/general/StaticView.ts]: > 

This throws in the error `This expression is not callable. Type 'never' has no call signatures.ts(2349)` for line 28. Using typeof also throws in error.

Comment (bwindels) [reply src/platform/web/ui/general/StaticView.ts]: Ok, if you can have a look if we can do better than `any` for this callback, they would be great :+1:

Comment (shubhangi013) [reply src/platform/web/ui/general/StaticView.ts]: > 

I'll look into it. I had to use `any` in both line 22 and 23 to avoid any error. :sweat_smile: So I can't think of assigning any other type to render.

**Comment by shubhangi013** [comment]
@bwindels I've done the suggested changes.

**Comment by shubhangi013** [comment]
@bwindels  could you review once? :sweat_smile:

**Comment by bwindels** [comment]
I'm sorry, but getting this into shape will take up too much of my time.
